packages/react-sdk/src/state/synchronizedDocumentImpl.ts

```diff
@@ -149,6 +149,11 @@ export class SynchronizedDocumentImpl<T extends Record<string, unknown>>
         // deduplicate the requests in case the snapshot creation is slower than 5000ms
         throttle(
           async ({ doc }) => {
+            if (!this.statistics.snapshotOutstanding) {
```

Comment:
Would it make sense to DRY this and just call the new `persist` method here? It's essentially the same code except for the try / catch.

Reply:
Good point, refactored :hammer_and_pick: